modules/generator/instance.go

```diff
@@ -261,14 +266,27 @@ func (i *instance) pushSpans(ctx context.Context, req *tempopb.PushSpansRequest)
 func (i *instance) updatePushMetrics(req *tempopb.PushSpansRequest) {
 	size := 0
 	spanCount := 0
+	expiredSpans := 0
 	for _, b := range req.Batches {
 		size += b.Size()
 		for _, ils := range b.InstrumentationLibrarySpans {
 			spanCount += len(ils.Spans)
+			// filter spans that have end time > max_age
+			var newSpansArr []*v1.Span
+			timeNow := time.Now().UnixNano()
+			for _, span := range ils.Spans {
+				if span.EndTimeUnixNano >= uint64(timeNow-i.cfg.MaxSpanAge*1000000000) {
+					newSpansArr = append(newSpansArr, span)
+				} else {
+					expiredSpans++
+				}
+			}
+			ils.Spans = newSpansArr
 		}
 	}
 	metricBytesIngested.WithLabelValues(i.instanceID).Add(float64(size))
 	metricSpansIngested.WithLabelValues(i.instanceID).Add(float64(spanCount))
+	metricSpansDiscarded.WithLabelValues(i.instanceID, "max_age_reached").Add(float64(expiredSpans))
```

Comment:
Also just a nitpick: consider making `max_age_reached` a constant. If we have multiple reasons, we can put them together and add some documentation about each.